### PR TITLE
@W-11464344@: Refactored tests for UnusedMethodRule.

### PR DESCRIPTION
**Pull request by jfeingold35**
This PR does the following:
- [WIP]

### CODE REVIEW

File: sfge/src/test/java/com/salesforce/rules/unusedmethod/CollidingNamesTest.java [hunk 1]
```diff
@@ -7,8 +7,12 @@
...
 import org.junit.jupiter.params.provider.ValueSource;
 
-/** A collection of tests for weird edge cases. */
-public class EdgeCaseTest extends BaseUnusedMethodTest {
+/**
+ * There are a bunch of weird edge cases that can happen when class names, variable names, and
+ * method names collide. These tests validate those. They are complicated enough to justify being
+ * their own file.
+ */
+public class CollidingNamesTest extends BaseUnusedMethodTest {
```
Comment (jfeingold35) [inline sfge/src/test/java/com/salesforce/rules/unusedmethod/CollidingNamesTest.java:15]: Renamed this test file to be more specific.

File: sfge/src/test/java/com/salesforce/rules/unusedmethod/MiddlemanTest.java [hunk 2]
```diff
@@ -1,67 +1,14 @@
...
-                    // property.
-                    + "    public static boolean b1 = DefiningClass.someMethod();\n"
-                    + "    public boolean b2 = DefiningClass.someOtherMethod();\n"
-                    + "}\n"
-        };
-        assertNoViolations(sourceCodes, 2);
-    }
-
-    /* =============== SECTION 2: INSTANCE METHODS =============== */
+public class MiddlemanTest extends BaseUnusedMethodTest {
```
Comment (jfeingold35) [inline sfge/src/test/java/com/salesforce/rules/unusedmethod/MiddlemanTest.java:11]: The instance method tests involving various middleman methods and properties were kept in this class, since in my opinion they're complicated enough (and verbose enough) to justify being in their own file.

**Review by rmohan20 — APPROVED** [review]
